### ISSUE TITLE
Add support for tokens inside ramp and animation

### ISSUE THREAD

**Issue by DRKblade** [polybar/polybar#2223]
## Is your feature request related to a problem? Please describe.
When I use the ramp node in a resource modules (such as CPU, Memory, Temp, Battery), I also want to see the information of the resource that it monitors, which is available in the tokens. So I add a label node to display the info.

The problem arises when I use different styles on different points of the ramp (like green for high battery, red for low battery). Now the styles on the label can't match the ramp, because the label can only have 1 style. 

## Why does polybar need this feature?
This would be an alternative to the feature requested in #1845 by using tokens in an animation node.

I think when people use ramp and animation nodes in resource modules, they most likely will use a label node to display the info. This feature would allow merging the two nodes, simplifying the config and potentially improving the performance.

This feature would allow the user to dynamically format the informational part of a module, instead of just the the ramp and animation.

## Describe the solution you'd like
I have successfully implement the feature locally, basically I add a config to specify a token string for all frames of the ramp/animation. When a frame is displayed, its content will be applied to the token `%label%` in the token string.

## Describe alternatives you've considered
I tried matching the label style to the ramp styles, which as stated above, is impossible since label have only 1 style while ramps have multiple styles.

**Comment by patrick96** [comment]
Yes, this is a good idea. I have mentioned this before in #1752 but I didn't create a separate issue for it.

The way I had in mind we could do this is that each module has a set of tokens (instead of each label) and that those tokens are available in all labels (the different parts of ramps and animations are also labels). Then you could do things like (cpu module):

```dosini
ramp-load-0 = %percentage%
ramp-load-1 = %percentage%
ramp-load-2 = %percentage%
ramp-load-2-background = #ff0000 
```

I don't think I completely understand the solution you have proposed. How would the `%label%` token be used?
